epsagon/wrappers/fastapi.py

```diff
@@ -48,26 +56,152 @@ def _handle_response(response, status_code, trace, raised_err):
     :return: the response, raising raised_err if its not None.
     """
     try:
-        if not raised_err and isinstance(response, Response):
+        if raised_err:
+            status_code = DEFAULT_ERROR_STATUS_CODE
+        elif isinstance(response, Response):
             if ignore_request(
                     response.headers.get('Content-Type', '').lower(),
                     ''
             ):
-                epsagon.trace.trace_factory.pop_trace(trace=trace)
+                epsagon_scope[SCOPE_IGNORE_REQUEST] = True
                 return response
 
         if response is not None:
             trace.runner.update_response(response, status_code=status_code)
-        epsagon.trace.trace_factory.send_traces()
+
     except Exception as exception:  # pylint: disable=W0703
-        print_debug('Failed to send traces: {}'.format(exception))
-        epsagon.trace.trace_factory.pop_trace(trace=trace)
+        print_debug('Failed to handle response: {}'.format(exception))
 
     if raised_err:
         raise raised_err
 
     return response
 
+
+def _get_epsagon_scope_data(request: Request) -> str:
+    """
+    Gets the epsagon scope data from a given request.
+    Returns None if not found.
+    """
+    try:
+        if not request.scope:
```

Comment:
well it shouldn't happen as its a part of the ASGI interface. Anyway, we check it so we will be failsafe (at this unexpected scenario it will be fine to collect a trace)